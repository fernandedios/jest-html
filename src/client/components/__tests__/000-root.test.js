/* eslint-env jest */
/* eslint-disable global-require, import/newline-after-import */
import React from 'react';
import renderer from 'react-test-renderer';
import Root from '../000-root';

// https://github.com/facebook/react/issues/7386#issuecomment-238091398
jest.mock('react-dom');
jest.mock('react-router', () => ({
  BrowserRouter: require('./mockComponent')('BrowserRouter'),
  Match: require('./mockComponent')('Match'),
  Miss: require('./mockComponent')('Miss'),
}));
jest.mock('../010-app', () => require('./mockComponent')('App'));
jest.mock('../200-largeMessage', () => require('./mockComponent')('LargeMessage'));

describe('Root', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Root />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
