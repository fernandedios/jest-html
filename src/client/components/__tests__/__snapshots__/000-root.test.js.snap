exports[`Root renders correctly 1`] = `
<div
  dataMockType="BrowserRouter">
  <div>
    <div
      component={[Function]}
      dataMockType="Match"
      exactly={true}
      pattern="/">
      Match
    </div>
    <div
      component={[Function]}
      dataMockType="Match"
      pattern="/suite/*">
      Match
    </div>
    <div
      component={[Function]}
      dataMockType="Match"
      pattern="/folder/*">
      Match
    </div>
    <div
      dataMockType="Miss"
      render={[Function]}>
      Miss
    </div>
  </div>
</div>
------------HTML PREVIEW---------------
  <div
    dataMockType="BrowserRouter">
    <div>
      <div
        dataMockType="Match"
        exactly
        pattern="/">
        Match
      </div>
      <div
        dataMockType="Match"
        pattern="/suite/*">
        Match
      </div>
      <div
        dataMockType="Match"
        pattern="/folder/*">
        Match
      </div>
      <div
        dataMockType="Miss">
        Miss
      </div>
    </div>
  </div>
`;
